sched/sched/sched_note.c

```diff
@@ -361,6 +421,13 @@ void sched_note_start(FAR struct tcb_s *tcb)
   int namelen;
 #endif
 
+#ifdef CONFIG_SCHED_INSTRUMENTATION_FILTER
+  if (!g_note_filter.enable)
```

Comment:
should we move this check to sched_note_add? so all duplication can be removed.

Reply:
I have wanted to reduce the performance affect as possible when the trace feature is not used. How about it?